projects/ngx-d3timeline/src/lib/activity.ts

```diff
@@ -7,3 +9,22 @@ export interface Activity {
   title?: string;
   sequence?: number;
 }
+
+const positionedActivityToActivityMap = {
+  updatedSeries: 'series',
+  updatedFinish: 'finish',
+  updatedStart: 'start'
+};
+
+export function getAcivityFromPositionedActivity(
+  positionedActivity: PositionedActivity
+): Activity {
+  const activity: Activity = { ...positionedActivity };
+
+  Object.keys(positionedActivityToActivityMap).forEach(k => {
```

Comment:
IMO this would be easier to understand if we just copied the values over individually. Could be missing something but not sure what is gained by using the mappings etc and I'm still not clear on what its doing.

Reply:
>  IMO this would be easier to understand if we just copied the values over individually

Then we would to update this function every time we added/removed properties.

> Could be missing something but not sure what is gained by using the mappings etc 

Just defines a mapping from positioned activity to activity